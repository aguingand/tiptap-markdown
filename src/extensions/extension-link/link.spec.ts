import { expect, test } from "vitest";
import { parse } from "../../../__tests__/utils";


test('parse markdown', () => {
    expect(parse('[example](http://example.org)')).toMatchSnapshot();
});
test('parse markdown with linkify', () => {
    expect(parse('http://example.org', { linkify:true })).toMatchSnapshot();
});
test('parse markdown without linkify', () => {
    expect(parse('http://example.org')).toMatchSnapshot();
});
test('parse html', () => {
    expect(parse('<a href="http://example.org">example</a>')).toMatchSnapshot();
});
