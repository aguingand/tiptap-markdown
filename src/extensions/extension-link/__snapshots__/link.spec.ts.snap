// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse html 1`] = `
[
  {
    "content": [
      {
        "marks": [
          {
            "attrs": {
              "class": null,
              "href": "http://example.org",
              "rel": "noopener noreferrer nofollow",
              "target": "_blank",
            },
            "type": "link",
          },
        ],
        "text": "example",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse markdown 1`] = `
[
  {
    "content": [
      {
        "marks": [
          {
            "attrs": {
              "class": null,
              "href": "http://example.org",
              "rel": "noopener noreferrer nofollow",
              "target": "_blank",
            },
            "type": "link",
          },
        ],
        "text": "example",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse markdown with linkify 1`] = `
[
  {
    "content": [
      {
        "text": "http://example.org",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse markdown without linkify 1`] = `
[
  {
    "content": [
      {
        "text": "http://example.org",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;
